src/Raven.Server/Rachis/Leader.cs

```diff
@@ -712,8 +712,13 @@ public void Initialize()
                 return BlittableResultWriter == null ? r : (r.Index, BlittableResultWriter.Result);
             } 
 
-            public void Dispose()
+            public async ValueTask DisposeAsync()
             {
+                if (Consumed.Raise() == false && Tcs.Task.IsCompleted == false)
+                    // If the command has already been dequeued, we must allow it to continue to maintain its context validity.
+                    // If we reached this point, another exception has already been thrown, and we want it to be the one that gets raised.
+                    await Tcs.Task.ContinueWith(_ => { }, TaskContinuationOptions.ExecuteSynchronously);
```

Comment:
I think there is a race here... if the task is already completed (a race with the line above) you might get a wrong exception.
why not just to wrap this with `try/catch` ?

Reply:
done